### DISCUSSION TITLE
500th error in the absence of the page's js file

### DISCUSSION THREAD

**Discussion by miker059** [nuxt/nuxt#17631] [Questions]
Hello, friends! 
There is a strange problem that pops up when crawling the page by search robots in SSR mode. 
The fact is that the robot indexes the page with the 500th error. 
That is, the server gives a normally rendered document, but at some point the 500th error occurs and everything collapses. As it was possible to find out the 500th error of such a plan - Failed to fetch dynamically imported module: http://0.0.0.0:3000/_nuxt/index-43e74fe1.mjs . 
index-43e74fe1.mjs  -this is the file of the page being opened.
At the same time, this error occurs randomly and only when crawled by search robots. It is possible to repeat this error only by forcibly blocking the download of the index-43e74fe1.mjs file. 
The nature of this error is not entirely clear and why, if this file is suddenly unavailable, then everything else loads normally on the client, but the 500th error is displayed? 

Does anyone have any guesses about this?

<img width="541" alt="image" src="https://user-images.githubusercontent.com/6391978/166670030-49f2b4d6-2c62-481e-9587-32313a95ae11.png">

<img width="1145" alt="image" src="https://user-images.githubusercontent.com/6391978/166670481-921f9aa6-992a-445b-b858-2605854ab6ba.png">

**Comment by miker059** [comment]
As I was able to understand, this problem may arise due to the fact that the mode of modern packages is always enabled during assembly. Changing the modern property in the configuration file does not lead to anything, even when I set the modern property to the fasle value, the assembly still occurs using modules. Apparently, search robots cannot process modern packages correctly yet. @danielroe tell me please, you are the plans to introduce support for the modern property?

**Reply by danielroe** [reply]
This is an interesting issue. Do you have any more information on the search engine that is unable to fetch `.mjs` files? The issue to follow for supporting legacy browsers is https://github.com/nuxt/framework/issues/13.

**Reply by miker059** [reply]
I'm not completely sure that the problem is that the search engine robot can't figure out the modules. Perhaps this is some kind of static cache on the side of the search engine, which leads to the fact that the index.mjs is not loaded and for some reason the page turns into the 500th error. Here I would rather like to understand how to make sure that the absence of some module did not cause the 500th error on the client with the 200th response code.

By the way, the search engine - yandex.com
